### ISSUE TITLE
Promises that return values now error on finished

### ISSUE THREAD

**Issue by corbinu** [hapijs/lab#700]
The latest change in 13.0.3 has caused tests that return a promise that returns an value to cause an error. Looks like it was working from 10.1.0 on.

```
 Non Error object received or caught

      at Immediate.finish (/home/corbinu/project/node_modules/lab/lib/runner.js:646:19)
      at runCallback (timers.js:676:20)
      at tryOnImmediate (timers.js:645:5)
      at processImmediate [as _immediateCallback] (timers.js:617:5)
```

for example a test
```
function getVals() {
    return new Promise((resolve) => {
        resolve([1, 2, 3]);
    });
}

lab.test("Returns value", () => {

    return getVals().then((vals) => {
        expect(vals).to.be.an.array();

        return vals;
    });

});
```

**Comment by Marsup** [comment]
Indeed, my bad. At least now we have a test for it :)

**Comment by corbinu** [comment]
No worries happy to help :)

**Comment by geek** [comment]
@corbinu thanks for the help, this is now released as v13.0.4

**Comment by jcollum** [comment]
This will also happen if you add `done` to a promise chain: 

```js
 Promise.mapSeries( //bluebird 
          [
            ... many promises
          ],
          () => {} //iterator, don't care about iteration here 
        )
          .then(done) // oops, done is called with the results of the mapSeries call!! 
          .catch(err => {
            console.error(err);
            done(err);
          });
      });

 ```

Solution: 

```js
 .then(() => done() )
```

That error is mighty cryptic though. Could it be changed to "done called with non-Error object"?
